File: capacity-club-api/src/domain-modules/workout/model/payload/workout-update.payload.ts
import { UniqueId } from '@common/model/unique-id';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, Length, IsString, IsOptional } from 'class-validator';
import { HomeTraining } from 'domain-modules/home-training/model/entity/home-training.entity';
import { IndoorTraining } from 'domain-modules/indoor-training/model/entity/indoor-training.entity';
import { TrainingCircuit } from 'domain-modules/training-circuit/model/entity/training-circuit.entity';

/**
 * Class used to update workout.
 * This class defines the payload used when updating existing workout.
 */
export class WorkoutUpdatePayload {
  @ApiProperty()
  @IsNotEmpty()
  @Length(26, 26)
  workout_id: UniqueId;
  @ApiProperty()
  @IsString()
  @IsNotEmpty()
  @Length(1, 50)
  title: string;
  @ApiProperty({ type: () => TrainingCircuit, isArray: true })
  @IsOptional()
  training_circuits: TrainingCircuit[];
  @ApiProperty({ type: () => HomeTraining })
  @IsOptional()
  home_training: HomeTraining;
  @ApiProperty({ type: () => IndoorTraining })
  @IsOptional()
  indoor_training: IndoorTraining;
}

File: capacity_club_mobile_app/lib/domain-features/data/workout/model/payload/workout_update_payload.dart
import 'package:capacity_club_mobile_app/core/model/abstract/payload.dart';
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';
import 'package:capacity_club_mobile_app/domain-features/data/home-training/model/home_training_model.dart';
import 'package:capacity_club_mobile_app/domain-features/data/indoor-training/model/indoor_training_model.dart';
import 'package:capacity_club_mobile_app/domain-features/data/training-circuit/model/training_circuit_model.dart';
import 'package:equatable/equatable.dart';

class WorkoutUpdatePayload extends Equatable implements Payload {
  final UniqueId workout_id;
  final String title;
  final List<TrainingCircuitModel> training_circuits;
  final IndoorTrainingModel indoor_training;
  final HomeTrainingModel home_training;

  const WorkoutUpdatePayload({
    required this.workout_id,
    required this.title,
    required this.training_circuits,
    required this.indoor_training,
    required this.home_training,
  });

  Map<String, dynamic> toJson() => {
        'workout_id': workout_id.toJson(),
        'title': title,
        'training_circuits':
            training_circuits.map((item) => item.toJson()).toList(),
        'indoor_training': indoor_training.toJson(),
        'home_training': home_training.toJson(),
      };

  @override
  List<Object?> get props =>
      [workout_id, title, training_circuits, indoor_training, home_training];
}
